README.md

```diff
@@ -1,39 +1,51 @@
 
-## Demo
 
-You have a [documented demo](https://lodr.github.io/offliner) in the `/demo` folder. 
+#### Demo
 
-The problem with the link before is _GitHub_ changes the protocol to **http** as soon as you access the page and you need to change it again to **https** for the demo to work.
+There is a [documented demo](https://lodr.github.io/offliner) in the `/demo` folder.  *Note: You will have to change the protocol to **https** in order for this demo link to work. (It automatically changes to **http** when you first access the link.)*
 
-To run locally, type:
+To run locally, clone this repo and run an `npm install` from the root directory. Then start a webserver with:
 
 ```bash
 $ gulp webserver
 ```
 
-And enter `demo/index.html` to see the demo in action. Try to shut the webserver down after visiting `index.html` and enter to some file explanation to make the **offliner** act.
+Navigate to `demo/index.html` to see the demo in action. Shut the webserver down after visiting `index.html`, and you should see that you are still able to access the page offline.
+
+If you edit one of the files (i.e. `/demo/js/offliner-fetcher-urls.js.html`) and restart the server, clicking the 'Check for updates' button will detect the new version, and allow you to download and activate the update.
+
+
+## The Web App Lifecycle
+Enabling an offline experience for your web application requires a few different steps that form the basis for an application lifecycle. This lifecycle consists of three distinct states:
+
+* **Installing:** Occurs on a user's first visit to your application. The installation phase prepares your application to be persistent and available for offline interaction. In this stage, you are provided with an offline cache where you can define the resources that must be available in order to provide an offline experience. 
 
-## Registering the worker
+* **Serving:** Occurs on any subsequent usage or interaction with your web application. All of the resources for offlining have already been cached and are fetched each time a user returns to the app.
+
+* **Updating:** Occurs when the developer iterates on the application and wants to push an updated version to users. The process for this stage occurs entails several steps:
+  1. **Check** for an update - determine the latest version and see if an update is needed
+  2. **Download** the update - populate a new cache to replace the current resources
+  3. **Activate** the update - replace the current cache with the new cache populated in the download step
+
+More information on how to implement this lifecycle with offliner is available below in the [Getting Started](https://github.com/brittanystoroz/offliner/tree/concept#getting-started) section.
```

Comment:
Replace the link with `#getting-started` and it will be relative to the document.